### ISSUE TITLE
Front-end tooling upgrades

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/guide#151]
Things I think we could do with:

- [x] Prettier setup
- [x] ESLint
- [x] Stylelint
- [ ] nvm -> fnm https://github.com/wagtail/guide/pull/185

---

De-scoped / future improvements split out

- [ ] #111
- [ ] #188
- [ ] #189

**Comment by lb-** [comment]
Thanks. I'd already created individual issues so I'll work on them bit by bit. 

I'll also move us to `npm` unless there is a huge pushback

**Comment by lb-** [comment]
Some PRs up - will let these get merged or get feedback and look at the other parts later.

* https://github.com/wagtail/guide/pull/159
* https://github.com/wagtail/guide/pull/156
* https://github.com/wagtail/guide/pull/159

**Comment by thibaudcolas** [comment]
🙌 I hadn’t realised there were existing issues

**Comment by thibaudcolas** [comment]
@lb- what do you think of splitting the CI build between frontend and backend checks once you revisit yarn/npm? Right now it looks like we’re running front-end linting tasks last even though they’re one of the fastest checks.

Here is a sample split setup in GitHub Actions if it helps: https://github.com/torchbox/storybook-django/blob/main/.github/workflows/ci.yml

**Comment by lb-** [comment]
@thibaudcolas  good call, we also did a similar thing for the Hallo package
https://github.com/wagtail/wagtail-hallo/blob/main/.github/workflows/ci.yml

**Comment by lb-** [comment]
https://github.com/wagtail/guide/pull/185 added for `fnm` usage. Once this is merged we can close this issue as the other items are a bit more nice to have and probably not blockers for release.
